imap/xapian_wrap.cpp

```diff
@@ -417,82 +415,73 @@ void xapian_check_if_needs_reindex(const strarray_t *sources, strarray_t *torein
 
 struct xapian_dbw
 {
-    Xapian::WritableDatabase *database;
-    Xapian::Stem *stemmer;
-    Xapian::TermGenerator *term_generator;
-    Xapian::Document *document;
+    Xapian::WritableDatabase database;
```

Comment:
I remember we ran into trouble passing C++ objects in this struct to the C code. That's why we used pointers. But I can't replicate that issue with my current setup, so probably it's due to and old compiler version we are using. I'll check.